app/controllers/attachments_controller.rb

```diff
@@ -0,0 +1,38 @@
+class AttachmentsController < ApplicationController
+  def new
+    @attachment = Attachment.new
+  end
+
+  def create
+    attachment = Attachment.new(params[:attachment])
+    document = fetch_document
+
+    if document.upload(attachment) && document.save!
+      flash[:success] = "Attached #{attachment.title}"
+      redirect_to edit_document_path(params[:document_type], document.content_id)
+    else
+      flash[:danger] = "There was an error uploading the attachment, please try again later."
+      redirect_to new_document_attachment_path(params[:document_type], document.content_id)
+    end
+  end
+
+  def edit
+
+  end
+
+  private
+  def fetch_document
+    begin
```

Comment:
There's no need to use explicit `begin`/`end` when it encompasses the whole method.